ci-operator/step-registry/trigger-jobs/trigger-jobs-commands.sh

```diff
@@ -7,66 +7,74 @@ SECRETS_DIR=/run/secrets/ci.openshift.io/cluster-profile
 GANGWAY_API_TOKEN=$(cat $SECRETS_DIR/gangway-api-token)
 WEEKLY_JOBS="$SECRETS_DIR/$JSON_TRIGGER_LIST"
 URL="https://gangway-ci.apps.ci.l2s4.p1.openshiftapps.com"
+sleep 1800
 
-echo "# Printing the jobs-to-trigger JSON:"
-jq -c '.[]' "$WEEKLY_JOBS"
-
-echo ""
-echo "# Test to make sure gangway api is up and running."
-max_retries=60
-retry_interval=60  # 60 seconds = 1 minute
-
-for ((retry_count=1; retry_count<=$max_retries; retry_count++)); do
-  response=$(curl -s -X GET -d '{"job_execution_type": "1"}' -H "Authorization: Bearer ${GANGWAY_API_TOKEN}" "${URL}/v1/executions/${PROW_JOB_ID}" -w "%{http_code}\n" -o /dev/null)
-
-  if [ "$response" -eq 200 ]; then
-    echo "Endpoint is up and returning HTTP status code 200 (OK)."
-    break  # Exit the loop if successful response received
-  else
-    echo "Endpoint is not available or returned an error (HTTP status code $response). Retrying..."
-  fi
-
-  # Sleep for the specified interval before the next retry
-  sleep $retry_interval
-done
-
-if [ "$response" -ne 200 ]; then
-  echo "Endpoint is still not available after $max_retries retries. Aborting."
-  exit 1
-fi
-
-max_retries=3
-failed_jobs=""
-
-echo ""
-echo "# Loop through the trigger weekly jobs file using jq and issue a command for each job where 'active' is true"
-jq -r '.[] | select(.active == true) | .job_name' "$WEEKLY_JOBS" | while IFS= read -r job_name; do
-  echo "Issuing trigger for active job: $job_name"
-  for ((retry_count=1; retry_count<=$max_retries; retry_count++)); do
-    response=$(curl -s -X POST -d '{"job_execution_type": "1"}' -H "Authorization: Bearer ${GANGWAY_API_TOKEN}" "${URL}/v1/executions/$job_name" -w "%{http_code}\n" -o /dev/null)
+# TODO: implement check to make sure we do not run the same build twice.
+$LATEST_BUILD_ID=$(curl -s "https://prow.ci.openshift.org/job-history/gs/origin-ci-test/logs/${LP_TRIGGER_JOB_NAME}" | grep -o 'var allBuilds = \[[^]]*' | awk -F '"' '{print $8}')
```

Comment:
@ascerra I found in my project that you can get the latest build by going to the GCP bucket for this job, and it will have a `latest-build.txt` in the root that you can use to get this. It will probably be more reliable than using the curl command.

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-rhpit-interop-tests-main-weekly_trigger-ocp-self-managed-layered-product-interop

Reply:
Great find I'll be using this :)